app/code/Magento/Downloadable/Observer/SaveDownloadableOrderItemObserver.php

```diff
@@ -131,6 +131,13 @@ public function execute(\Magento\Framework\Event\Observer $observer)
                         ScopeInterface::SCOPE_STORE
                     );
                 $linkPurchased->setLinkSectionTitle($linkSectionTitle)->save();
+                
+                $linkstatus =  \Magento\Downloadable\Model\Link\Purchased\Item::LINK_STATUS_PENDING;
+               if( $orderStatusToEnableItem == \Magento\Sales\Model\Order\Item::STATUS_PENDING
+                    || $orderItem->getOrder()->getStatus() == \Magento\Sales\Model\Order::STATE_COMPLETE){
```

Comment:
I guess we should use the `getState` instead of `getStatus` method in this case.

Reply:
sure